### ISSUE TITLE
Add application/x-www-form-urlencoded body parsing

### ISSUE THREAD

**Issue by bdearborn** [l2ig/aqua#31]
This can be done pretty easily by adding some code to **parseBody**:

_(Please ignore the changes I made to get it working in JS and extracting the body)_

```
function parseBody(req) {

    var textDecoder = new TextDecoder();
    const rawBody = textDecoder.decode(req.raw.r.buf);
    let body = {};
  
    if (!rawBody) return {};
  
    try {
      body = JSON.parse(rawBody);
    }catch(error) {
        if (rawBody.includes('name="')) { //multipart/form-data
            body = (rawBody.match(/name="(.*?)"(\s|\n|\r)*(.*)(\s|\n|\r)*---/gm) || [])
                .reduce((fields, field) => {
                    if (!/name="(.*?)"/.exec(field)?.[1]) return fields;
  
                    return {
                        ...fields,
                        [/name="(.*?)"/.exec(field)?.[1] || ""]: field.match(/(.*?)(?=(\s|\n|\r)*---)/)?.[0]
                    }
                }, {});
        } else { //application/x-www-form-urlencoded
            const urlParamsStr = rawBody.substr(rawBody.indexOf('\r\n\r\n')+4);
            body = Object.fromEntries(new URLSearchParams(urlParamsStr))
        }
    }
  
    return body;
  }
```
This is the part that does the x-www-form-urlencode parsing:

```
        } else { //application/x-www-form-urlencoded
            const urlParamsStr = rawBody.substr(rawBody.indexOf('\r\n\r\n')+4);
            body = Object.fromEntries(new URLSearchParams(urlParamsStr))
        }
```

**Comment by l2ig** [comment]
Hey, thanks for your contribution! The code looks good to me, I'll just write some tests to make sure and release a new version afterwards.

**Comment by bdearborn** [comment]
Thank you!

**Comment by l2ig** [comment]
Landed in v1.0.14. Thanks again for your help! :)
